### PR TITLE
Chore: Refactored daily-challenge

### LINKED ISSUE #425: Convert !challenge to new event system

### PR DESCRIPTION
**Pull request by hasanlatch007**
The `READY` event would be used as a kickstart and never used again unless the bot was stopped when `PostDailyChallenge` was being executed and stopped before it could run the `scheduleTimer`, then it will be used again
Also in case, this fails added a command to be able to reset it.
This !closes #425

### CODE REVIEW

**Comment by hasanlatch007** [comment]
Just realized i forgot to set the time correctly wait for the new commit before reviewing

File: src/programs/daily-challenge.ts [hunk 1]
```diff
@@ -1,128 +1,271 @@
...
-      await messageChannel.send(embed);
+      await DailyChallengeChannel.send("@group dailychallenge");
+      await DailyChallengeChannel.send(embed);
+
+      const executeTime = new Date();
+      executeTime.setUTCHours(8, 0, 0);
+      executeTime.setDate(executeTime.getDate() + 2);
+      await TimerService.createTimer(dailyChallengeIdentifier, executeTime, {
+        channelId: DailyChallengeChannel.id,
+      });
```
Comment (geisterfurz007) [inline src/programs/daily-challenge.ts:123]: This code fragment has been duplicated twice in this file; stay [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

**Review by geisterfurz007 — CHANGES_REQUESTED** [review]

File: src/programs/daily-challenge.ts [hunk 2]
```diff
@@ -77,52 +156,98 @@ export const postDailyMessage = async (
...
+
+        await DailyChallengeChannel.send("@group dailychallenge");
+        await DailyChallengeChannel.send(embed);
+
+        const executeTime = new Date();
+        executeTime.setUTCHours(8, 0, 0);
+        executeTime.setDate(executeTime.getDate() + 2);
+        await TimerService.createTimer(dailyChallengeIdentifier, executeTime, {
+          channelId: DailyChallengeChannel.id,
+        });
```
Comment (geisterfurz007) [inline src/programs/daily-challenge.ts:199]: Dupe

File: src/programs/daily-challenge.ts [hunk 3]
```diff
@@ -77,52 +156,98 @@ export const postDailyMessage = async (
...
+        return;
       }
     } else {
-      const res = { result: info };
-      await save(res, message);
-      await message.react("👍");
+      await message.react("👎");
+      logger.error(
+        "Failed to reset dailly challenge due to res being null (theres no errors to log good luck!)"
+      );
```
Comment (geisterfurz007) [inline src/programs/daily-challenge.ts:211]: No worries about that; it would blow up in line 157 before getting here :)

**Review by geisterfurz007 — CHANGES_REQUESTED** [review]

**Comment by hasanlatch007** [comment]
blocked until https://github.com/Yes-Theory-Fam/database/pull/40 is merged

**Comment by geisterfurz007** [comment]
Merged *and* published! Upgrade database to 0.14.0 for the fix

File: src/programs/daily-challenge.ts [hunk 4]
```diff
@@ -141,3 +160,16 @@ const save = async (
...
+const startDailyChallengeTimer = async (identifier: string) => {
+  const msDay = 24 * 60 * 60 * 1000;
+  const executeTime = new Date();
+  let executionDate = 2;
+  if (Math.floor(Date.now() / msDay) % 2) {
+    executionDate = 1;
+  }
+
+  executeTime.setUTCHours(UTC_HOUR_POSTED, 0, 0);
+  executeTime.setMinutes(executeTime.getDate() + executionDate);
```
Comment (geisterfurz007) [inline src/programs/daily-challenge.ts:173]:
```suggestion
  executeTime.setDate(executeTime.getDate() + executionDate);
```

**Review by geisterfurz007 — CHANGES_REQUESTED** [review]

**Review by geisterfurz007 — APPROVED** [review]
:shipit:

**Pull request merged**